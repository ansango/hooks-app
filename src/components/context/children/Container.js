
import "./Container.css";
export const Container = ({ title = "", children }) => {
  return (
    <div className="base-container py-5 bg-light">
      <h3 className="text-center">{title}</h3>
      <div className="px-2">{children}</div>
    </div>
  );
};
