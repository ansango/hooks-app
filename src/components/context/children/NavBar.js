import { NavLink } from "react-router-dom";

export const NavBar = ({ routes }) => {
  return (
    <nav className="navbar navbar-light bg-light d-flex flex-row justify-content-end">
      <div className="navbar-nav">
        <div className="d-flex">
          {routes.map((route) => (
            <NavLink
              key={route.name}
              exact
              activeClassName="active"
              className="nav-item nav-link mx-3"
              to={route.path}
            >
              {route.name}
            </NavLink>
          ))}
        </div>
      </div>
    </nav>
  );
};
