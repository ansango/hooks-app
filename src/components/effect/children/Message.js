import { useEffect, useState } from "react";

export const Message = () => {
  const [coords, setCoords] = useState({ x: 0, y: 0 });
  const { x, y } = coords;
  useEffect(() => {
    const mouseMove = (e) => setCoords({ x: e.x, y: e.y });
    window.addEventListener("mousemove", mouseMove);
    return () => {
      window.removeEventListener("mousemove", mouseMove);
    };
  }, []);
  return (
    <div>
      X: <p>{x}</p>
      Y: <p>{y}</p>
    </div>
  );
};
