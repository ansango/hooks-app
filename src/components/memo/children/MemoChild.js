import React from "react";

export const MemoChild = React.memo(({ number, increment }) => {
  console.log("render");

  return (
    <button className="btn btn-primary me-3" onClick={() => increment(number)}>
      {number}
    </button>
  );
});
