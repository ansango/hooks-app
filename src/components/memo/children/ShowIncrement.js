import React from "react";
import { log } from "../../../helpers/log";

export const ShowIncrement = React.memo(({ increment }) => {
  log("ShowIncrement", "Render");
  return (
    <button className="btn btn-primary" onClick={() => increment(5)}>
      Incrementar
    </button>
  );
});
