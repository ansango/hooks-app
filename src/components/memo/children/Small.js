import React from "react";
import { log } from "../../../helpers/log";

export const Small = React.memo(({ value }) => {
  log("Small", "Render");
  return <small>{value}</small>;
});
