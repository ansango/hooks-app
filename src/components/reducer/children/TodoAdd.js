import { generateId } from "../../../helpers/generateId";
import { useForm } from "../../../hooks/useForm";

export const TodoAdd = ({ handleAddTodo }) => {
  const [{ description }, handleInputChange, reset] = useForm({
    description: "",
  });

  const handleSubmit = (e) => {
    e.preventDefault();
    if (description.trim().length !== 0) {
      const newTodo = {
        id: generateId(),
        description: description,
        done: false,
      };

      handleAddTodo(newTodo);
      reset();
    }
  };
  return (
    <>
      <h3> Add Todo</h3>
      <hr />
      <form onSubmit={handleSubmit} className="py-2">
        <input
          type="text"
          placeholder="Add a todo"
          name="description"
          autoComplete="off"
          className="form-control mb-3"
          value={description}
          onChange={handleInputChange}
        />
        <button className="btn btn-outline-primary w-100" type="submit">
          Add
        </button>
      </form>
    </>
  );
};
