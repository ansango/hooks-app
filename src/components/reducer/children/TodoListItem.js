import "./TodoListItem.css";

export const TodoListItem = ({
  todo,
  index,
  handleDeleteTodo,
  handleToggleTodo,
}) => {
  return (
    <li className="list-group-item d-flex items-center justify-content-between">
      <p
        className={`${todo.done && "complete"}`}
        onClick={() => handleToggleTodo(todo.id)}
      >
        {index + 1}. {todo.description}
      </p>
      <button
        className="btn btn-danger"
        onClick={() => handleDeleteTodo(todo.id)}
      >
        Delete
      </button>
    </li>
  );
};
