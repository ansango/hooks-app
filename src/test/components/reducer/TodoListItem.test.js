import { shallow } from "enzyme";
import { TodoListItem } from "../../../components/reducer/children/TodoListItem";
import { todos } from "../../fixtures/todos";
describe("Test in <TodoListItem/>", () => {
  const handleDeleteTodo = jest.fn();
  const handleToggleTodo = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      todo={todos[0]}
      index={0}
      handleDeleteTodo={handleDeleteTodo}
      handleToggleTodo={handleToggleTodo}
    />
  );
  test("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should call delete function", () => {
    wrapper.find("button").simulate("click");
    expect(handleDeleteTodo).toHaveBeenCalledWith(todos[0].id);
  });
  test("should call toggle function", () => {
    wrapper.find("p").simulate("click");
    expect(handleToggleTodo).toHaveBeenCalledWith(todos[0].id);
  });

  test("should has class complete if todo is done", () => {
    todos[0].done = true;
    const wrapper = shallow(<TodoListItem todo={todos[0]} index={0} />);
    expect(wrapper.find("p").hasClass("complete")).toBe(true);
  });
  test("should render correct text", () => {
    const p = wrapper.find("p");
    expect(p.text()).toBe(`${0 + 1}. ${todos[0].description}`);
  });
});
