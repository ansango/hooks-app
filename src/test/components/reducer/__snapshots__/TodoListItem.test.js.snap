// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <TodoListItem/> should render 1`] = `
<li
  className="list-group-item d-flex items-center justify-content-between"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    Learn React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
